### ISSUE TITLE
Renaming Mod Placed Train Stops Causes Crash

### ISSUE THREAD

**Issue by legendblade** [Yousei9/Logistic-Train-Network#102]
If a mod places a train stop, renaming that train stop will cause LTN to crash at https://github.com/Yousei9/Logistic-Train-Network/blob/master/control.lua#L509 due to the fact that the entity was never placed into the old name list unless the mod explicitly fires `defines.events.on_built_entity` or `defines.events.on_robot_built_entity`

**Comment by Yousei9** [comment]
That's bad practice by that mod.
I can't add special handling to everything other mods do wrong.

**Comment by legendblade** [comment]
While I'm open to the idea that I was in the wrong by not firing off what appears to be the wrong events when placing down entities via script/mod (the docs suggest that http://lua-api.factorio.com/latest/events.html#script_raised_built would be the proper event in this case, but given its nature, I can understand why it's not being handled), immediately dismissing the issue by saying that it's bad practice for a mod to not fire one of those two specific events is a little rough.

I've already fixed the code on my end so that it raises the 'proper' build event based on the type of the original event - but I'd question how mods that aren't building stations as a direct result of player/robot building are supposed to handle letting your mod know that a station was built since, in those cases, raising `on_built_entity`/`on_robot_built_entity` would be wrong as there's no source player/robot and they can't just assume that every other mod won't require one of those two properties to be set.

Either way, I believe that the underlying bug I pointed out is something that LTN should fix, as your code is just assuming that a station always exists in your internal map and then trying to index that station.  Obviously if you choose not to fix it it doesn't bother me - as I've already said, I've fixed what was causing it on my end - but given that the stacktrace will point at LTN as being the mod crashing, it's probably just going to result in other issues like this cropping up.

Sorry for bothering you.
